client/sources/ok_test/models.py

```diff
@@ -100,9 +100,11 @@ def unlock(self, interact):
                                 for case in suite.cases])
         for suite_num, suite in enumerate(self.suites):
             for case_num, case in enumerate(suite.cases):
+                case_id = '{} > Suite {} > Case {}'.format(
```

Comment:
The data format described in the doc looks correct to me (for what we've currently agreed on).

@soumya, those two cases would record the following 4 attempts:
```
{'prompt': '>>> x = 5\n>>>square(x)',   # first prompt of first case
 'answer': '25'},
{'prompt': '>>> square(-x)',   # second prompt of first case
 'answer': '25'},
{'prompt': '>>> x = -5\n>>>square(x)',   # first prompt of second case
 'answer': '25'},
{'prompt': '>>> square(-x)',   # second prompt of second case
 'answer': '25'},
```
You'll notice that attempt 2 and attempt 4 are identical; this occurs because attempts are recorded for each prompt within a test case (attempts are not recorded for the test case over all).

@ksteph your stated purpose of the hash aligns with my understanding of it. From what @soumyabasu described, where `ID = hash(assignment_name, test_name, prompt)` is insufficient, because it's possible for the same prompt to show up more than once in a test (see example I gave above).

My point about removing the `case_id` is that there's no reason to have both a `case_id` and a `hash`, since they're supposed to do the same thing.